sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionUtils.java

```diff
@@ -27,7 +46,19 @@
      * @throws Exception If the underlying reflective calls throw an exception.
      */
     public MethodHandles.Lookup getLookupToUse(Class<?> targetClass) throws Exception {
-        return MethodHandles.publicLookup();
+        try {
+            return (MethodHandles.Lookup) PRIVATE_LOOKUP_IN.invoke(targetClass);
+        } catch (Throwable throwable) {
+            if (throwable instanceof Error) {
+                throw (Error) throwable;
+            }
+
+            if (throwable instanceof Exception) {
+                throw (Exception) throwable;
+            }
+
+            throw new Exception(throwable);
+        }
```

Comment:
Is this catch block necessary?

Reply:
As in catching itself or the style of the catch block?

Reply:
Catching itself since we are rethrowing Error or Exception without logging. So, we can remove this catch block and let the original `Throwable` bubble up.

Reply:
The issue is that the API checks with `Exception` so the `Throwable` needs to be handled. I can add logging for the exception, though it should only be called from locations that will handle logging if an exception is thrown.

Reply:
Discussed, confusion was that `.invoke()` throws `Throwable`. Changed the pattern slightly as there should only ever be `Error`s or `Exception`s with the APIs being invoked.